### DISCUSSION TITLE
Render to fetch request?

### DISCUSSION THREAD

**Discussion by noconsulate** [diegomura/react-pdf#2365] [Q&A] [answered]
I want my react app to render a PDF and then send it off via a fetch request. Is this possible with react-pdf and can someone point me in the right direction? 

Thanks

**Comment by CoskunCakir** [comment, marked as answer]
hey @noconsulate, i think you could send the blob data off via a fetch req, so it could be something like this?:

```JS
const blob = pdf(<MyDocument />).toBlob();

blob.then((blob) => {
  fetch('https://endpoint', {
    method: 'POST',
    body: blob,
  })
  .then((response) => response.json())
  .then((data) => console.log(data))
  .catch((error) => console.log(error));
});
```

also, you can check here: https://react-pdf.org/advanced#on-the-fly-rendering

**Comment by noconsulate** [comment]
Wow, this is great. Just what I needed to put me in the right path. Thanks!
